keras/optimizers/schedules/learning_rate_schedule.py

```diff
@@ -641,6 +641,14 @@ def __init__(
         self.decay_steps = decay_steps
         self.alpha = alpha
         self.name = name
+        
+        #validating decay_steps argument to raise error if passed
+        #negative values or zero or float values or float Tensors
+        if isinstance(self.decay_steps,tf.Tensor):
+            self.decay_steps = self.decay_steps.numpy()
```

Comment:
Note that the tensor here could be a graph tensor, so the numpy() method might raise an error. Also there is no point to convert a tensor back to numpy and then convert back to tensor again in __call__.